src/spatialdata/_core/spatialdata.py

```diff
@@ -541,16 +541,23 @@ def transform_element_to_coordinate_system(
         The transformed element.
         """
         from spatialdata import transform
+        from spatialdata.transformations import Identity
         from spatialdata.transformations.operations import (
             get_transformation_between_coordinate_systems,
+            remove_transformation,
+            set_transformation,
         )
 
         t = get_transformation_between_coordinate_systems(self, element, target_coordinate_system)
         transformed = transform(element, t, maintain_positioning=False)
+        remove_transformation(transformed, remove_all=True)
+        set_transformation(transformed, Identity(), target_coordinate_system)
+
         return transformed
```

Comment:
why does it set it and returns it? also, should not create transformed just for return but return directly `transform(element, t, maintain_positioning=False)`

Reply:
The transform function preserves all the mapping to coordinate systems. Here we explicitly remove them. Two comments. After the transformation refactoring I plan to move this function, and `transform_to_coordinate_system()` somewhere else, or maybe replace them with `transform()`. So I keep it like this, but next week we should revisit this.